CoreScriptsRoot/Modules/Settings/Pages/Players.lua

```diff
@@ -117,12 +134,42 @@ local function Initialize()
 					if friendLabel and friendLabelText and friendLabelText.Text ~= '' then
 						friendLabel.ImageTransparency = 1
 						friendLabelText.Text = ''
-						if localPlayer and player then
+					end
+					if AtFriendLimit(localPlayer) then
+						while PromptCreator:IsCurrentlyPrompting() do
+							wait()
+						end
+						PromptCreator:CreatePrompt({
+							WindowTitle = "Friend Limit Reached",
+							MainText = string.format("You can not send a friend request to %s because you are at the max friend limit.", player.Name),
+							ConfirmationText = "Okay",
+							CancelActive = false,
+							Image = BUST_THUMBNAIL_URL ..player.UserId,
+							ImageConsoleVR = THUMBNAIL_URL ..player.UserId,
+							StripeColor = Color3.fromRGB(183, 34, 54),
+						})
+						friendStatusCreate(playerLabel, player)
```

Comment:
This seems unnecessary. It looks like the friendLabel was already cleared earlier in this code path.